source/jormungandr/jormungandr/equipments/sytral.py

```diff
@@ -97,34 +108,84 @@ def record_call(self, status, **kwargs):
         params.update(kwargs)
         new_relic.record_custom_event('parking_status', params)
 
-    def _process_data(self, data, stop_points_list):
+    def _fill_equipment_details(self, equipment_form_web_service, equipment_details):
+        equipment_details.id = equipment_form_web_service['id']
+        equipment_details.name = equipment_form_web_service['name']
+        equipment_details.embedded_type = type_pb2.EquipmentDetails.EquipmentType.Value(
+            '{}'.format(equipment_form_web_service['embedded_type'])
+        )
+        equipment_details.current_availability.status = type_pb2.CurrentAvailability.EquipmentStatus.Value(
+            '{}'.format(equipment_form_web_service['current_availaibity']['status'])
+        )
+        current_availaibity = equipment_form_web_service['current_availaibity']
+        for period in current_availaibity['periods']:
+            p = equipment_details.current_availability.periods.add()
+            p.begin = date_to_timestamp(parser.parse(period['begin']))
+            p.end = date_to_timestamp(parser.parse(period['end']))
+        equipment_details.current_availability.updated_at = current_availaibity['updated_at']
+        equipment_details.current_availability.cause.label = current_availaibity['cause']['label']
+        equipment_details.current_availability.effect.label = current_availaibity['effect']['label']
+
+    def _fill_default_equipment_details(self, id, embedded_type, equipment_details):
+        equipment_details.id = id
+        equipment_details.embedded_type = type_pb2.EquipmentDetails.EquipmentType.Value(embedded_type)
+        equipment_details.current_availability.status = type_pb2.CurrentAvailability.EquipmentStatus.Value(
+            "unknown"
+        )
+
+    def _embedded_type(self, embedded_type_sytral):
```

Comment:
Can you not use the `embedded_type` from SytralRT response instead ?

What type is the `embedded_type_sytral` ? If you work with an Enum, you can get the name with Enum.value.name field.

Reply:
it corresponds in proto to 
```
message Code {
    required string type = 1;
    required string value = 2;
}
``
and it is a string, so no Enum is possible because it is provided directly from the response...